File: easy_1/point-mutations/ruby/v2/point_mutations.rb
# point_mutations.rb

class DNA
  def initialize(strand)
    self.strand = strand
  end

  def hamming_distance(other_strand)
    if strand.length <= other_strand.length
      shorter = strand
      longer = other_strand
    else
      shorter = other_strand
      longer = strand
    end

    count = 0
    shorter.each_char.with_index do |char, idx|
      count += 1 if char != longer[idx]
    end

    count
  end

  protected
  attr_accessor :strand
end

File: easy_1/point-mutations/js/point_mutations.js
class DNA {
  constructor(strand) {
    this.strand = strand;
  }

  hammingDistance(otherStrand) {
    const iterationDuration = this.determineIterationDuration(this.strand, otherStrand);
    let distance = 0;

    for(let idx = 0; idx < iterationDuration; idx += 1) {
      if(this.strand[idx] !== otherStrand[idx]) {
        distance += 1;
      }
    }
    return distance;
  }

  determineIterationDuration(strand, otherStrand) {
    if (strand.length <= otherStrand.length) {
      return strand.length;
    } else {
      return otherStrand.length;
    }
  }
}

export default DNA;